disnake/onboarding.py

```diff
@@ -0,0 +1,239 @@
+        self.prompts = [
+            OnboardingPrompt._from_dict(data=prompt, guild=self.guild) for prompt in data["prompts"]
+        ]
+        self.enabled = data["enabled"]
+        self._default_channel_ids = list(map(int, data["default_channel_ids"]))
+
+    def __repr__(self) -> str:
+        return (
+            f"<Onboarding guild_id={self.guild.id!r} prompts={self.prompts!r}"
+            f" default_channel_ids={self._default_channel_ids!r} enabled={self.enabled!r}>"
+        )
+
+    @property
+    def default_channels(self) -> List[GuildChannel]:
+        """List[:class:`abc.GuildChannel`]: The list of channels that will be shown to new members by default."""
+        return list(filter(None, map(self.guild.get_channel, self._default_channel_ids)))
+
+
+class OnboardingPromptOption(Hashable):
+    """Represents an onboarding prompt option.
+
+    .. versionadded:: 2.9
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The prompt option's ID.
+    emoji: Optional[Union[:class:`PartialEmoji`, :class:`Emoji`]]
+        The prompt option's emoji.
+    title: :class:`str`
+        The prompt option's title.
+    description: :class:`str`
+        The prompt option's description.
+    """
+
+    __slots__ = ("id", "title", "description", "emoji", "guild", "_roles_ids", "_channels_ids")
+
+    def __init__(
+        self,
+        *,
+        title: str,
+        description: str,
+        emoji: Optional[Union[str, PartialEmoji, Emoji]],
+        roles: Sequence[Snowflake],
+        channels: Sequence[Snowflake],
+        guild: Guild,
+    ):
+        # NOTE: The ID may sometimes be a UNIX timestamp since
+        # Onboarding changes are saved locally until you send the API request (that's how it works in client)
+        # so the API needs the timestamp to know what ID it needs to create, should we just add a note about it
+        # or "try" to create the ID ourselves?
+        # I'm not sure if this also happens for OnboardingPrompt
+        self.id = 0
+        self.title = title
+        self.description = description
+        self.guild = guild
+        self._roles_ids = [role.id for role in roles]
+        self._channels_ids = [channel.id for channel in channels]
+
+        self.emoji: Optional[Union[PartialEmoji, Emoji]] = None
+        if emoji is None:
+            self.emoji = None
+        elif isinstance(emoji, str):
+            self.emoji = PartialEmoji.from_str(emoji)
+        elif isinstance(emoji, _EmojiTag):
+            self.emoji = emoji
+        else:
+            raise TypeError("emoji must be a str, PartialEmoji, or Emoji instance")
+
+    def __str__(self) -> str:
+        return self.title
+
+    def __repr__(self) -> str:
+        return (
+            f"<OnboardingPromptOption id={self.id!r} title={self.title!r} "
+            f"description={self.description!r} emoji={self.emoji!r} "
+            f"roles={self.roles!r} channels={self.channels!r}>"
```

Comment:
Don't think this should contain `roles` and `channels`, they're also rather verbose

Reply:
dae932be51d4522f81a40a11a416f251f3bc2354